test/extended/networking/isolation.go

```diff
@@ -1,35 +1,74 @@
 	"k8s.io/kubernetes/test/e2e"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
-// This test requires a network plugin that supports namespace isolation.
-// NOTE: if you change the test description, update networking.sh too!
 var _ = Describe("[networking] network isolation plugin", func() {
 	f1 := e2e.NewFramework("net-isolation1")
 	f2 := e2e.NewFramework("net-isolation2")
 
-	It("should prevent communication between pods in different namespaces", func() {
-		By("Picking multiple nodes")
-		nodes := getMultipleNodes(f1)
-		node1 := nodes.Items[0]
-		node2 := nodes.Items[1]
+	It("should prevent communication between pods in different namespaces on the same node", func() {
+		if pluginIsolatesNamespaces() {
+			checkPodIsolation(f1, f2, false)
+		} else {
+			e2e.Skipf("Network plugin does not implement isolation.")
+		}
+	})
 
-		By("Running a webserver in one namespace")
-		podName := "isolation-webserver"
-		defer f1.Client.Pods(f1.Namespace.Name).Delete(podName, nil)
-		ip := launchWebserverPod(f1, podName, node1.Name)
+	It("should prevent communication between pods in different namespaces on different nodes", func() {
+		if pluginIsolatesNamespaces() {
+			checkPodIsolation(f1, f2, true)
+		} else {
+			e2e.Skipf("Network plugin does not implement isolation.")
+		}
+	})
 
-		By("Checking that the webserver is not accessible from a pod in a different namespace on the same node")
-		err := checkConnectivityToHost(f2, node1.Name, "isolation-same-node-wget", ip, 10)
-		Expect(err).To(HaveOccurred())
+	It("should allow communication between pods in different namespaces on the same node", func() {
+		if pluginIsolatesNamespaces() {
+			e2e.Skipf("Network plugin implements isolation.")
+		} else {
+			checkPodIsolation(f1, f2, false)
+		}
+	})
 
-		By("Checking that the webserver is not accessible from a pod in a different namespace on a different node")
-		err = checkConnectivityToHost(f2, node2.Name, "isolation-different-node-wget", ip, 10)
-		Expect(err).To(HaveOccurred())
+	It("should allow communication between pods in different namespaces on different nodes", func() {
+		if pluginIsolatesNamespaces() {
+			e2e.Skipf("Network plugin implements isolation.")
+		} else {
+			checkPodIsolation(f1, f2, true)
+		}
 	})
 })
+
+func checkPodIsolation(f1, f2 *e2e.Framework, differentNodes bool) {
+	nodes, err := e2e.GetReadyNodes(f1)
+	if err != nil {
+		e2e.Failf("Failed to list nodes: %v", err)
+	}
+	var serverNode, clientNode *api.Node
+	serverNode = &nodes.Items[0]
+	if differentNodes {
+		if len(nodes.Items) == 1 {
+			e2e.Skipf("Only one node is available in this environment")
+		}
+		clientNode = &nodes.Items[1]
+	} else {
+		clientNode = serverNode
+	}
+
+	podName := "isolation-webserver"
+	defer f1.Client.Pods(f1.Namespace.Name).Delete(podName, nil)
+	ip := e2e.LaunchWebserverPod(f1, podName, serverNode.Name)
+
+	err = checkConnectivityToHost(f2, clientNode.Name, "isolation-wget", ip, 10)
+	if pluginIsolatesNamespaces() {
```

Comment:
Consider passing this value in to the function to avoid unnecessary coupling to the test environment.